src/nvim/generators/gen_api_ui_events.lua

```diff
@@ -0,0 +1,164 @@
+    proto_output:write(';\n')
+
+    if not ev.remote_impl then
+      remote_output:write('static void remote_ui_'..ev.name)
+      write_signature(remote_output, ev, 'UI *ui')
+      remote_output:write('\n{\n')
+      write_arglist(remote_output, ev, true)
+      remote_output:write('  push_call(ui, "'..ev.name..'", args);\n')
+      remote_output:write('}\n\n')
+    end
+
+    if not ev.bridge_impl then
+      bridge_output:write('static void ui_bridge_'..ev.name)
+      write_signature(bridge_output, ev, 'UI *ui')
+      bridge_output:write('\n{\n')
+
+      argv, recv, recv_argv, recv_cleanup = '', '', '', ''
+      argc = 1
+      for j = 1, #ev.parameters do
+        local param = ev.parameters[j]
+        copy = 'copy_'..param[2]
+        if param[1] == 'String' then
+          bridge_output:write('  String copy_'..param[2]..' = copy_string('..param[2]..');\n')
+          argv = argv..', '..copy..'.data, INT2PTR('..copy..'.size)'
+          recv = (recv..'  String '..param[2]..
+                          ' = (String){.data = argv['..argc..'],'..
+                          '.size = (size_t)argv['..(argc+1)..']};\n')
+          recv_argv = recv_argv..', '..param[2]
+          recv_cleanup = recv_cleanup..'  api_free_string('..param[2]..');\n'
+          argc = argc+2
+        elseif param[1] == 'Array' then
+          bridge_output:write('  Array copy_'..param[2]..' = copy_array('..param[2]..');\n')
+          argv = argv..', '..copy..'.items, INT2PTR('..copy..'.size)'
+          recv = (recv..'  Array '..param[2]..
+                          ' = (Array){.items = argv['..argc..'],'..
+                          '.size = (size_t)argv['..(argc+1)..']};\n')
+          recv_argv = recv_argv..', '..param[2]
+          recv_cleanup = recv_cleanup..'  api_free_array('..param[2]..');\n'
+          argc = argc+2
+        elseif param[1] == 'Integer' or param[1] == 'Boolean' then
+          argv = argv..', INT2PTR('..param[2]..')'
+          recv_argv = recv_argv..', PTR2INT(argv['..argc..'])'
+          argc = argc+1
+        else
+          assert(false)
+        end
+      end
+      bridge_output:write('  UI_CALL(ui, '..ev.name..', '..argc..', ui'..argv..');\n}\n')
+      bridge_output:write('static void ui_bridge_'..ev.name..
+                          '_event(void **argv)\n{\n')
+      bridge_output:write('  UI *ui = UI(argv[0]);\n')
+      bridge_output:write(recv)
+      bridge_output:write('  ui->'..ev.name..'(ui'..recv_argv..');\n')
+      bridge_output:write(recv_cleanup)
+      bridge_output:write('}\n\n')
+    end
+  end
+
+  call_output:write('void ui_call_'..ev.name)
+  write_signature(call_output, ev, '')
+  call_output:write('\n{\n')
+  if ev.remote_only then
+    write_arglist(call_output, ev, false)
+    call_output:write('  ui_event("'..ev.name..'", args);\n')
+  else
+    call_output:write([[
+  flush_cursor_update();
+  for (size_t i = 0; i < ui_count; i++) {
+    UI *ui = uis[i];
```

Comment:
it was the plan at first, but then I realized it was best to keep it. I will change this back.